improver/nbhood/square_kernel.py

```diff
@@ -196,13 +215,23 @@ def run(self, cube: Cube, radius: float, mask_cube: Optional[Cube] = None) -> Cu
             Cube containing the smoothed field after the square
             neighbourhood method has been applied.
         """
+        check_if_grid_is_equal_area(cube)
+
         # If the data is masked, the mask will be processed as well as the
         # original_data * mask array.
         check_radius_against_distance(cube, radius)
         original_attributes = cube.attributes
         original_methods = cube.cell_methods
         grid_cells = distance_to_number_of_grid_cells(cube, radius)
         nb_size = 2 * grid_cells + 1
+        if self.neighbourhood_method == "circular":
+            axes = []
+            for axis in ["x", "y"]:
+                coord_name = cube.coord(axis=axis).name()
+                axes.append(cube.coord_dims(coord_name)[0])
+
+            self.kernel = circular_kernel(grid_cells, self.weighted_mode)
```

Comment:
```suggestion
            self.kernel = circular_kernel(grid_cells, self.weighted_mode)
        elif self.neighbourhood_method == "square":
            self.nb_size = 2 * grid_cells + 1
        else:
            raise Your Chosen Exception for an unrecognised neighbourhood_method.
```
The second elif will allow you to modify `_calculate_neighbourhood` to remove nb_size as a required argument, which it isn't if we are using a circular kernel. The else allows for an exception if the chosen neighbourhood method is not valid, as I can't see that checked in the plugin, which I think it should be.

Reply:
Done, I think I may check whether this is the right place for this to be raised in a future PR as part of sorting out the base class stuff.